hack/e2e/utils.sh

```diff
@@ -52,3 +52,31 @@ kill_podman() {
         fi
     fi
 }
+
+setup_environment() {
+    echo "########## 🌐 Setting up Azure account and secrets ##########"
+    az account set -s "$AZURE_SUBSCRIPTION_ID"
+    SECRET_SA_ACCOUNT_NAME="$SECRET_SA_ACCOUNT_NAME" make secrets
+    . secrets/env
+    export CI=true
```

Comment:
idk if this works as expected or not, I would imagine that we would want to configure this at the pipeline level as a variable of some sort so it is guaranteed to be set across jobs/etc? This also has the risk of potentially being executed locally, and doing things like deleting clusters or creating clusters that are not needed in local flows.

Reply:
We did a lot of debugging and confirmed that make secrets needs to be invoked in each containerized step.